### ISSUE TITLE
A k8s service backed by a virtual service cannot act as destination in another virtual service.

### ISSUE THREAD

**Issue by arunpmohan** [istio/istio#30662]
I have a setup where I have a place holder k8s service called  demo-service which do not have any selectors. Now there is a virtual service attached to this demo service  so that any request coming to demo-service should be routed to demo1-service and demo2-service based on port or any kind of route rule.

All my internal microservices call this demo-service and everything seems to work.

Now i have another virtual service which is binded to an ingress gateway and which has to do some URI prefix matching to send traffic to this demo-service. When i do that i always get 503 . IF i replace it with actual service having endpoints then it works . 

So i have 2 modes of traffic and i am trying to centralize the logic for routing rules in my-demo-vrs.

Traffic flows 

**1.  Istio Ingress ==> ingress-vrs(demo-service) ==> my-demo-vrs(demo-service) ==> demo1-service or demo2-service ==> 503 response .** 
2.  Mesh internal micro services ==> my-demo-vrs(demo-service) ==> demo1-service or demo2-service==> works fine.

Here are my sample files.

**Demo Service without selectors**

```
apiVersion: v1
kind: Service
metadata:
  name: demo-service
spec:
  clusterIP: 10.98.148.45
  ports:
  - name: http-8080
    port: 8080
    protocol: TCP
    targetPort: 8080
  - name: http-8081
    port: 8081
    protocol: TCP
    targetPort: 8081
  - name: http2-8082
    port: 8082
    protocol: TCP
    targetPort: 8082
```

**Virtual Service which routes to demo service.Used for internal mesh communication from other micro services**

```
kind: VirtualService
metadata:
  name: my-demo-vrs
spec:
  exportTo:
  - .
  hosts:
  - demo-service
  http:
  - match:
    - port: 8080
    route:
    - destination:
        host: demo1-service
        port:
          number: 8080
  - match:
    - port: 8082
    route:
    - destination:
        host: demo2-service
        port:
          number: 8082
  - match:
    - port: 8081
    route:
    - destination:
        host: demo-service
        port:
          number: 8081
```

**Second virtual service which is binded to an http gateway for prefix based routing**

```
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: ingress-vrs
spec:
  exportTo:
  - .
  gateways:
  - shared-istio-gateway
  hosts:
  - mysite.web.com
  http:
  - match:
    - uri:
        prefix: /a/demo1
    - uri:
        prefix: /test/hello1
      route:
        - destination:
            host: demo-service
            port:
              number: 8080
  - match:
    - uri:
        prefix: /b/demo2
    - uri:
        prefix: /test/hello3
    route:
    - destination:
        host: demo-service
        port:
          number: 8081
```

**Comment by howardjohn** [comment]
Maybe you want to use https://istio.io/latest/docs/reference/config/networking/virtual-service/#Delegate here

**Comment by arunpmohan** [comment]
Thanks a lot and this seems like what i need. Is this available in 1.5.10 istio ?

Also i see this article https://discuss.istio.io/t/try-to-create-a-delegate-virtual-service-but-got-error-configuration-invalid-virtual-service-must-have-at-least-one-host/7133.  and tried it but doesnt seem to work.
